psa-crypto/src/types/key.rs

```diff
@@ -305,6 +305,37 @@ impl Attributes {
         get_attributes_res?;
         Ok(attributes?)
     }
+
+    /// Sufficient size for a buffer to export the key, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn export_key_output_size(self) -> Result<usize> {
+        match self.key_type {
+            Type::RsaKeyPair
+            | Type::RsaPublicKey
+            | Type::EccKeyPair { .. }
+            | Type::EccPublicKey { .. } => Ok(psa_crypto_sys::PSA_EXPORT_KEY_OUTPUT_SIZE(
+                self.key_type.try_into()?,
+                self.bits,
+            )),
+            _ => Err(Error::NotSupported),
+        }
+    }
+
+    /// Sufficient buffer size for a signature using the given key, if the key is supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn sign_output_size(self) -> Result<usize> {
+        match self.key_type {
```

Comment:
I think here you can re-use the [`compatible_with_alg`](https://docs.rs/psa-crypto/0.2.0/psa_crypto/types/key/struct.Attributes.html#method.compatible_with_alg) method of `Attributes` which implements similar logic and do something like:
```rust
self.compatible_with_alg(alg)?;
Ok(PSA_SIGN_OUTPUT_SIZE(..))
```